### PR TITLE
Add tests for Popover

### LINKED ISSUE #468: Add tests for the Popover component
The popover has no tests. To convert it into a functional component, it is better to write tests before.

### PR DESCRIPTION
**Pull request by ritikamotwani**
Fixes: #468

### CODE REVIEW

File: src/Display/Popover/Popover.test.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+
+import * as renderer from 'react-test-renderer';
```
Comment (rosslo) [inline src/Display/Popover/Popover.test.tsx]: eslint warning here 🙏

File: src/Display/Popover/Popover.test.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  test('content prop should be visible when popover clicked once', () => {
+    const { queryByTestId, queryByText } = render(
+      <Popover content={<PopoverContent />}>
+        <button key="button"> Open Popover </button>
+      </Popover>
+    );
+
+    const contentTextBeforeClick = queryByText('Popover Content');
+    expect(contentTextBeforeClick).toBeNull();
```
Comment (rosslo) [inline src/Display/Popover/Popover.test.tsx]: Can consinder using `not.toBeInTheDocument()` instead. It's more descriptive to me.

File: src/Display/Popover/Popover.test.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    expect(onPopoverClick).not.toHaveBeenCalled();
+
+    const children = queryByTestId('children');
+    fireEvent.click(children);
```
Comment (rosslo) [inline src/Display/Popover/Popover.test.tsx]: I think we should use `user-event` as official recommendation https://testing-library.com/docs/guide-events#alternatives

File: src/Display/Popover/Popover.tsx [hunk 4]
```diff
@@ -56,7 +56,11 @@ class Popover extends React.Component<Props, State> {
         aria-busy="false"
         tabIndex={0}
       >
-        <PopoverChildren onClick={this.handleOnClick}>
+        <PopoverChildren
+          className="popover-children"
+          data-testid="children"
```
Comment (rosslo) [inline src/Display/Popover/Popover.tsx]: why do we need `className="popover-children"` if we have `data-testid="children"`?

File: src/Display/Popover/Popover.test.tsx (hunk 1 above)
Comment (ritikamotwani) [reply src/Display/Popover/Popover.test.tsx]: Removed. No idea why my editor didn't show a warning.

File: src/Display/Popover/Popover.test.tsx (hunk 3 above)
Comment (ritikamotwani) [reply src/Display/Popover/Popover.test.tsx]: Thanks. This is useful.

**Comment by ritikamotwani** [comment]
@rosslo , updated it. 🙏

**Review by rosslo — APPROVED** [review]

**Pull request merged**